plugins/microsoft_teams/icon_microsoft_teams/util/teams_utils.py

```diff
@@ -351,6 +398,37 @@ def get_message_from_chat(  # noqa: C901
     return remove_null_and_clean(message)
 
 
+def list_messages_from_chat(connection: insightconnect_plugin_runtime.connection, chat_id: str) -> list:  # noqa: C901
+    """
+    Lists the last 50 messages from a teams chat
+
+    :param connection: Object (insightconnect_plugin_runtime.connection)
+    :param chat_id: String
+    :return: boolean
+    """
+
+    params = {"$top": 50, "$orderby": "createdDateTime desc"}
+    message_url = f"https://graph.microsoft.com/beta/{connection.tenant_id}/chats/{chat_id}/messages/"
+    headers = connection.get_headers()
+    response = requests.get(message_url, headers=headers, params=params)  # nosec B113
+
+    try:
+        response.raise_for_status()
+    except Exception as error:
+        raise PluginException(preset=PluginException.Preset.BAD_REQUEST, data=error)
+    try:
+        messages = response.json().get("value", [])
+    except Exception as error:
+        raise PluginException(preset=PluginException.Preset.INVALID_JSON, data=error)
```

Comment:
would it be nicer to have:

``` 
try:
    response.raise_for_status()
    messages = response.json().get("value", [])
except HTTPError as http_error:
    raise PluginException(preset=PluginException.Preset.BAD_REQUEST, data=http_error)
except ValueError as value_error:
    raise PluginException(preset=PluginException.Preset.INVALID_JSON, data=value_error)
except Exception error:
    raise PluginException(preset=PluginException.Preset.UNKNOWN, data=error)
```

Reply:
This has been updated along side a small changed to the unit test to correctly test the http request handling